fpdf/image_parsing.py

```diff
@@ -127,24 +132,160 @@ def get_img_info(img, image_filter="AUTO", dims=None):
         ):
             info["smask"] = _to_data(img, image_filter, select_slice=alpha_channel)
 
+    dp = f"/Predictor 15 /Colors {dpn} /Columns {w}"
+    
+    if img.mode == "1":
+        inverted = False
+        # copied from img2pdf
+        if (
+            img.format == "TIFF"
+            and img.info["compression"] == "group4"
+            and len(img.tag_v2[TiffImagePlugin.STRIPOFFSETS]) == 1
+            and len(img.tag_v2[TiffImagePlugin.STRIPBYTECOUNTS]) == 1
+        ):
+            photo = img.tag_v2[TiffImagePlugin.PHOTOMETRIC_INTERPRETATION]
+            if photo == 0:
+                inverted = True
+        if inverted:
+            dp = f"/BlackIs1 true /Columns {w} /K -1 /Rows {h}"
+        else:
+            dp = f"/BlackIs1 false /Columns {w} /K -1 /Rows {h}"
+
     info.update(
         {
             "w": w,
             "h": h,
             "cs": colspace,
             "bpc": bpc,
             "f": image_filter,
-            "dp": f"/Predictor 15 /Colors {dpn} /Columns {w}",
+            "dp": dp,
         }
     )
 
     return info
 
+class temp_attr:
+    """
+    Copied verbatim from img2pdf
+
+    temporary change the attribute of an object using a context manager
+    """
+    def __init__(self, obj, field, value):
+        self.obj = obj
+        self.field = field
+        self.value = value
+
+    def __enter__(self):
+        self.exists = False
+        if hasattr(self.obj, self.field):
+            self.exists = True
+            self.old_value = getattr(self.obj, self.field)
+        print(f"setting {self.obj}.{self.field} = {self.value}")
+        setattr(self.obj, self.field, self.value)
+
+    def __exit__(self, exctype, excinst, exctb):
+        if self.exists:
+            setattr(self.obj, self.field, self.old_value)
+        else:
+            delattr(self.obj, self.field)
+
+def ccitt_payload_location_from_pil(img):
+    """
+    Copied verbatim from img2pdf
+    """
+    # If Pillow is passed an invalid compression argument it will ignore it;
+    # make sure the image actually got compressed.
+    if img.info["compression"] != "group4":
+        raise ValueError(
```

Comment:
Could you add a unit test for this case please?

Reply:
I'm not sure that I can, I don't think the code can go through this path... I'll remove the test

Reply:
OK